### Первая часть задания: Кноверсия
Для начала посчитаем доверительный интервал для нашей конверсии, после чего проверим его на статистическую значимость

Сначала сделаем необходимые import'ы библиотек

In [1]:
import math
import numpy as np
import pandas as pd

In [2]:
def is_stat_important(q1, q2, n1, n2):
    right_bound = q2 + (1.96 * math.sqrt(q2) * (1 - q2)) / n2
    left_bound = q2 - (1.96 * math.sqrt(q2) * (1 - q2)) / n2
    right_bound_diff = (q2 - q1) + 1.96 * math.sqrt((q1*(1-q1))/n1 + (q2*(1-q2))/n2)
    left_bound_diff = (q2 - q1) - 1.96 * math.sqrt((q1*(1-q1))/n1 + (q2*(1-q2))/n2)
    print(f'Confidence interval for conversion: [{left_bound}; {right_bound}]')
    print(f'Confidence interval of difference: [{left_bound_diff}; {right_bound_diff}]')
    return np.sign(right_bound_diff) == np.sign(left_bound_diff)

In [3]:
stat_importance_val = "Our estimation IS statistical important" \
        if is_stat_important(0.12123, 0.12466, 100000, 100000)\
        else "Our estimation IS NOT statistical important"
print(stat_importance_val)

Confidence interval for conversion: [0.12465394245837239; 0.12466605754162763]
Confidence interval of difference: [0.0005517136632461343; 0.006308286336753871]
Our estimation IS statistical important


Как видно из наших вычилсений, новая конверсия будет лешать в интервале от $[0.12465394245837239; 0.12466605754162763]$. \
Также, видим что наша оценка является статистически значимой, т.к. $0 \not\in [0.0005517136632461343; 0.006308286336753871]$

__Вывод__: т.к. наша оценка получилась статистически значимой, и интвервал больше нуля, то значит что группа $B$ показала лучшие результаты, и можно рассмотреть изменения дальше

### Вторая часть: Инкрементальная оценка инициатив 
Теперь посчитаем общую статистику по пользователям и покупателям

Для расчета данных по какой-то когорте

In [4]:
def print_df_data(df: pd.DataFrame, name='') -> None:
    buyers = df['AOV'].astype(bool).sum(axis=0)
    users = df['ID'].astype(bool).sum(axis=0)
    buyers_data = df.loc[(df['AOV'].astype(bool) != False)]
    avg_payment_cnt = buyers_data.loc[:, 'AVG PAYMENTS COUNT'].mean()
    sum_payment = buyers_data.loc[:, 'AVG PAYMENTS COUNT'].sum()
    avg_pay = buyers_data.loc[:, 'AOV'].mean()
    print(f'DF name: {name}')
    print(f'Users: {users}\tBuyers: {buyers}\tAOV: {avg_pay}\tPayments: {sum_payment}\tavg_payment_cnt: {avg_payment_cnt}')


In [11]:
def analysis_for_group():
    df = pd.read_csv('DECEMBER_DECEMBER', sep=',', header=0)
    print_df_data(df, name='DECEMBER_DECEMBER')

    df = pd.read_csv('DECEMBER_JANUARY', sep=',', header=0)
    print_df_data(df, name='DECEMBER_JANUARY')

    df = pd.read_csv('JANUARY_JANUARY', sep=',', header=0)
    print_df_data(df, name='JANUARY_JANUARY')

In [12]:
analysis_for_group()

DF name: DECEMBER_DECEMBER
Users: 1000000	Buyers: 120000	AOV: 2499.7765458764616	Payments: 240054.0	avg_payment_cnt: 2.00045
DF name: DECEMBER_JANUARY
Users: 417000	Buyers: 50000	AOV: 2700.046894243662	Payments: 99912.0	avg_payment_cnt: 1.99824
DF name: JANUARY_JANUARY
Users: 1250000	Buyers: 150000	AOV: 2999.3103936139355	Payments: 299958.0	avg_payment_cnt: 1.99972


__Вывод__: Как видно из Инкрементальной оценки инициатив(см. отчет), в случае принятия модели В, наше количество покупателей, возрастет на 5717,81

### Третья часть: TOTAL GMV
Все данные уже собраны, теперь осталось посчитать GMV по когортам, и для модели $B$. \
В отчете это все предоставленно, поэтому сдесь будет написан только вывод

__Вывод__: Как видно из отчета, в случае принятие данной модели, наш доход возрастет на 44 495 605,31 или же на 3.46%. \
Однако стоит понимать, что это не чистая прибыль поскольку нужно еще вычесть ФОТ, и уже потом мы полуим чистую прибыль. 
B итоге, поскольку разница такая колосальная, имеет смысл принять модель $B$